lib/service/kinveyCompletionHandler.js

```diff
@@ -12,16 +12,46 @@
  * the License.
  */
 
-function createCompletionHandler(task, callback) {
-  function completionHandler(entity) {
-    if (!entity) entity = {};
+function createCompletionHandler(task, updateRequestBody, callback) {
+  function completionHandler(body) {
+    if (typeof updateRequestBody === 'function') {
+      callback = updateRequestBody;
+      updateRequestBody = false;
+    }
 
     const responseCallback = callback;
-    const result = task.response;
+    const result = {};
 
-    result.body = entity;
+    function normalizeError(error) {
```

Comment:
Is it worth putting this function in its own module since it's used in multiple locations?